src/charm.py

```diff
@@ -233,6 +240,51 @@ def _deploy_k8s_resources(self) -> None:
     def _get_data_from_profiles_interface(self, kf_profiles_interface):
         return list(kf_profiles_interface.get_data().values())[0]
 
+    def _get_sidebar_items(self) -> List[SidebarItem]:
+        """Returns a list of all SidebarItems for this charm.
+
+        Includes sidebar items defined through relations and user config.
+        """
+        sidebar_items = []
+        sidebar_items.extend(self.sidebar_provider.get_sidebar_items())
+        sidebar_items.extend(self._get_sidebar_items_from_config())
+        return sidebar_items
+
+    def _get_sidebar_items_as_json(self) -> str:
+        """Returns a list of all SidebarItems for this charm, as a JSON string.
+
+        Includes sidebar items defined through relations and user config.
+        """
+        return sidebar_items_to_json(self._get_sidebar_items())
+
+    def _get_sidebar_items_from_config(self) -> List[SidebarItem]:
+        """Returns a list of SidebarItems as defined by the additional-sidebar-links config.
+
+        If there are errors in parsing the config, this returns an empty list and logs a warning.
+        """
+        error_message = (
+            f"Cannot parse user-defined sidebar links from config "
+            f"`{ADDITIONAL_SIDEBAR_LINKS_CONFIG}` - ignoring this input."
+        )
+
+        sidebar_config = self.model.config[ADDITIONAL_SIDEBAR_LINKS_CONFIG]
+        if not sidebar_config:
+            return []
+
+        try:
+            user_sidebar_links = yaml.safe_load(sidebar_config)
+        except Exception as err:  # TODO: Make this the right error
```

Comment:
would this be yaml.YAMLError ?

Reply:
good call!  Didn't realise that is a parent to all the errors it could raise.  ty!